modules/045-snapshot-controller/hooks/generate_webhook_certs.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2022 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/deckhouse/deckhouse/go_lib/certificate"
+)
+
+const (
+	namespace   = "kube-system"
+	serviceName = "snapshot-validation-webhook"
+	serviceHost = serviceName + "." + namespace + ".svc"
+	secretName  = "d8-snapshot-validation-webhook-certs"
+	certPath    = "snapshotController.internal.webhookCert"
+)
+
+type CertSnapshot struct {
+	Name string
+	Cert certificate.Certificate
+}
+
+func applyCertsFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	secret := &v1.Secret{}
+	err := sdk.FromUnstructured(obj, secret)
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert selfsigned certificate secret to secret: %v", err)
+	}
+
+	return CertSnapshot{
```

Comment:
Prefer a pointer here and later on when type asserting. Lots of stuff to copy lots of time for no reason at all.